src/expr/impl/src/scalar/vector.rs

```diff
@@ -521,3 +521,60 @@ fn l2_norm(vector: VectorRef<'_>) -> F64 {
 fn l2_normalize(vector: VectorRef<'_>) -> VectorVal {
     vector.normalized()
 }
+
+#[derive(Debug)]
+pub struct SubvectorContext {
+    pub start: i32,
+    pub end: i32,
+}
+
+impl SubvectorContext {
+    pub fn from_start_count(start: i32, count: i32) -> Result<Self> {
+        Ok(Self {
+            start,
+            end: start + count,
+        })
+    }
+}
+
+/// ```slt
+/// query R
+/// SELECT subvector('[1,2,3,4,5]'::vector(5), 1, 3);
+/// ----
+/// [1,2,3]
+///
+/// query R
+/// SELECT subvector('[1,2,3,4,5]'::vector(5), 3, 2);
+/// ----
+/// [3,4]
+///
+/// query R
+/// SELECT subvector('[1,2,3,4,5]'::vector(5), 1, 5);
+/// ----
+/// [1,2,3,4,5]
+///
+/// query R
+/// SELECT subvector('[1,2,3,4,5]'::vector(5), 5, 1);
+/// ----
+/// [5]
+///
+/// query R
+/// SELECT subvector('[1,2,3,4,5]'::vector(5), 2, 3);
+/// ----
+/// [2,3,4]
+/// ```
+#[function(
+    "subvector(vector, int4, int4) -> vector",
+    prebuild = "SubvectorContext::from_start_count($1, $2)?",
+    type_infer = "unreachable"
+)]
+fn subvector(v: VectorRef<'_>, ctx: &SubvectorContext) -> Result<VectorVal> {
+    let vector = v.as_slice();
+
+    let result = vector[(ctx.start - 1) as usize..(ctx.end - 1) as usize]
```

Comment:
Seems that we can store `start` and `end` directly as the start and end of slice range, so that we can directly use `vector[ctx.start..ctx.end]`.